### PR TITLE
[processor/k8sattributes] Add support for profiles signal

### LINKED ISSUE #35983: [processor/k8sattributes] Add support for profiles signal
### Component(s)

processor/k8sattributes

### Describe the issue you're reporting

This processor would be useful for testing and validation of the experimental profiling signal, and to replace some functionality available in the eBPF Agent.

As inspiration for how to add support this PR can be used: open-telemetry/opentelemetry-collector/pull/11071 (there is no processor with support as of the writing of this issue)

Example of configuration using profiles (needs `service.profilesSupport` [feature gate](https://github.com/open-telemetry/opentelemetry-collector/blob/main/featuregate/README.md#controlling-gates) to be enabled):

```yaml
receivers:
  otlp:
    protocols:
      grpc:
exporters:
  otlp:
    endpoint: ${OTLP_ENDPOINT}
service:
  pipelines:
    profiles:
      receivers: [otlp]
      exporters: [otlp]
```

### PR DESCRIPTION
**Pull request by haoqixu**
#### Description
Add support for profiles signal to `k8sattributesprocessor`.

#### Link to tracking issue
Fixes #35983

#### Testing
- factory_test.go
- processor_test.go

### CODE REVIEW

File: processor/k8sattributesprocessor/factory.go [hunk 1]
```diff
@@ -128,6 +141,36 @@ func createLogsProcessorWithOptions(
...
+
+	profilesConsumer, err := consumerprofiles.NewProfiles(func(ctx context.Context, pd pprofile.Profiles) (err error) {
+		pd = kp.processProfiles(ctx, pd)
+		return nextProfilesConsumer.ConsumeProfiles(ctx, pd)
+	}, consumer.WithCapabilities(consumerCapabilities))
+	if err != nil {
+		return nil, err
+	}
+
+	return &profiles{
```
[the remark this answers is not in the document]
Comment (mx-psi) [reply processor/k8sattributesprocessor/factory.go:167]: Oh, seems like I got confused on https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35983#issuecomment-2437644676, sorry. I am in favor of having a helper (which could live as internal API here or we can have `processorhelperprofiles`)

[the remark this answers is not in the document]
Comment (haoqixu) [reply processor/k8sattributesprocessor/factory.go:167]: I have submitted a PR (open-telemetry/opentelemetry-collector#11556) to add the `processorhelperprofiles` pkg and will update this PR after open-telemetry/opentelemetry-collector#11556 is merged.

**Review by dmathieu — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]
This LGTM, but I would like to do the update-otel separately, so all core modules point to the same commit

**Comment by haoqixu** [comment]
> This LGTM, but I would like to do the update-otel separately, so all core modules point to the same commit

The `go.opentelemetry.io/collector/receiver/receivertest` package has been moved into a separate module on the `main` branch and does not have a release tag yet. Running `multimod` will always fail on components depending on `receivertest`. :thinking:

**Review by ChrsMark — COMMENTED** [review]
Can we cover profiles in the e2e testing as well? See https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.112.0/processor/k8sattributesprocessor/e2e_test.go. Either tuning the existing tests or introducing a new one.

**Comment by haoqixu** [comment]
> Can we cover profiles in the e2e testing as well? See https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.112.0/processor/k8sattributesprocessor/e2e_test.go. Either tuning the existing tests or introducing the a new one.

I have updated the e2e test. However, the `telemetrygen` CLI doesn't support profile signals yet. I will submit another PR to add support for profile signals to `telemetrygen` and then update this PR.

**Comment by haoqixu** [comment]
> I will submit another PR to add support for profile signals to `telemetrygen` and then update this PR.

I have opened an issue (#36127) to track it and skip the tests with added `TODO`s. I would like to submit another PR to enable the profiles test and remove all `TODO`s from the tests once `telemetrygen` is ready for profiles.

**Comment by haoqixu** [comment]
> > This LGTM, but I would like to do the update-otel separately, so all core modules point to the same commit
> 
> The `go.opentelemetry.io/collector/receiver/receivertest` package has been moved into a separate module on the `main` branch and does not have a release tag yet. Running `multimod` will always fail on components depending on `receivertest`. 🤔

Hi, @mx-psi. I have rebased this PR to `main`, and the `main` branch has already updated `core` to `v1.19.0/0.113.0`.

**Review by TylerHelmuth — APPROVED** [review]
@fatsheep9146 @dmitryax please review.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**